### PR TITLE
Creation of VM with SIRF built

### PR DESCRIPTION
**Pull request by paskino**
This Vagrantfile is able to create a VM, and build&install SIRF. 
CTests are run and positive.

Commands needed:

```
vagrant up
vagrant provision
```

Known issues:
 1. https://github.com/CCPPETMR/CCPPETMR_VM/issues/39

### CODE REVIEW

**Comment by paskino** [comment]
https://github.com/CCPPETMR/CCPPETMR_VM/commit/904280602e710dad48d2c42752befe48e4837bfd fixes #39

**Comment by KrisThielemans** [comment]
I'm getting
```
PS C:\Users\krisf\Documents\devel\CCPPETMR_VM\vagrant> vagrant up
Bringing machine 'default' up with 'virtualbox' provider...
==> default: Box 'box-cutter/ubuntu1604' could not be found. Attempting to find and install...
    default: Box Provider: virtualbox
    default: Box Version: >= 0
==> default: Loading metadata for box 'box-cutter/ubuntu1604'
    default: URL: https://vagrantcloud.com/box-cutter/ubuntu1604
==> default: Adding box 'box-cutter/ubuntu1604' (v2.0.26) for provider: virtualbox
    default: Downloading: https://vagrantcloud.com/box-cutter/boxes/ubuntu1604/versions/2.0.26/providers/virtualbox.box
    default:
An error occurred while downloading the remote file. The error
message, if any, is reproduced below. Please fix this error and try
again.

The requested URL returned error: 404 Not Found
```
on my Windows 10 laptop with freshly installed vagrant, but I guess this is not our fault.

File: vagrant/Vagrantfile [hunk 1]
```diff
@@ -91,8 +120,11 @@ Vagrant.configure("2") do |config|
     export CCMAKE="ccmake -DCMAKE_PREFIX_PATH:PATH=$INSTALL_DIR/lib/cmake -DCMAKE_INSTALL_PREFIX:PATH=$INSTALL_DIR"
 
     git clone https://github.com/CCPPETMR/CCPPETMR_VM.git
-    bash CCPPETMR_VM/scripts/INSTALL_prerequisites_with_apt-get.sh
-    #bash CCPPETMR_VM/scripts/UPDATE.sh
+	cd CCPPETMR_VM
+	git pull
+	git checkout v0.9.2
+    bash $SRC_PATH/CCPPETMR_VM/scripts/INSTALL_prerequisites_with_apt-get.sh
```
Comment (KrisThielemans) [inline vagrant/Vagrantfile]: why explicit paths?

**Comment by KrisThielemans** [comment]
More info on the download error. This isn't a vagrant error, but the URL is indeed not accessible, e.g. from Firefox. With [trusty64](https://app.vagrantup.com/ubuntu/boxes/trusty64/versions/20171213.0.1) I don't have download problems.

I think the [16.04](https://app.vagrantup.com/box-cutter/boxes/ubuntu1604/versions/2.0.26) is out of date and not maintained. This seems to [have happened in the past](https://github.com/boxcutter/ubuntu/issues/133). probably means we need to have our own box anyway. Sigh. @bathomas, can you help?

File: vagrant/Vagrantfile [hunk 2]
```diff
@@ -38,6 +38,17 @@ Vagrant.configure("2") do |config|
...
     vb.memory = "4096"
+	
+	vb.customize "pre-boot", [
+       "storageattach", :id,
+       "--storagectl", "IDE Controller",
+        "--port", "1",
+       "--device", "0",
+       "--type", "dvddrive",
+       "--medium", "emptydrive",
+	   "--clipboard" , "bidirectional"
```
[the remark this answers is not in the document]
Comment (KrisThielemans) [reply vagrant/Vagrantfile]: that doesn't work. Splitting it off as a new configure command does seem to
```
	vb.customize "pre-boot", [
       "storageattach", :id,
       "--storagectl", "IDE Controller",
        "--port", "1",
       "--device", "0",
       "--type", "dvddrive",
       "--medium", "emptydrive"]
   vb.customize "pre-boot", [   
	   "modifyvm", :id,
	   "--clipboard" , "bidirectional"
       ]
```

**Comment by KrisThielemans** [comment]
more on the box version. Can we use https://app.vagrantup.com/ubuntu/boxes/xenial64 instead? 

Note that then we need to set
```
"--storagectl", "IDE",
```
i.e. removing the `Controller`. This seems to be because the [control name can change](https://github.com/hashicorp/vagrant/issues/8105)

File: vagrant/Vagrantfile [hunk 3]
```diff
@@ -91,8 +120,11 @@ Vagrant.configure("2") do |config|
     export CCMAKE="ccmake -DCMAKE_PREFIX_PATH:PATH=$INSTALL_DIR/lib/cmake -DCMAKE_INSTALL_PREFIX:PATH=$INSTALL_DIR"
 
     git clone https://github.com/CCPPETMR/CCPPETMR_VM.git
-    bash CCPPETMR_VM/scripts/INSTALL_prerequisites_with_apt-get.sh
-    #bash CCPPETMR_VM/scripts/UPDATE.sh
+	cd CCPPETMR_VM
+	git pull
+	git checkout v0.9.2
+    bash $SRC_PATH/CCPPETMR_VM/scripts/INSTALL_prerequisites_with_apt-get.sh
+    bash $SRC_PATH/CCPPETMR_VM/scripts/UPDATE.sh
```
Comment (KrisThielemans) [inline vagrant/Vagrantfile]: next line needs to be
```
chown -R sirfuser:users /home/sirfuser/
```
Otherwise various files are still owned by root

**Comment by KrisThielemans** [comment]
when using https://app.vagrantup.com/ubuntu/boxes/xenial64, it doesn't start the X server, I think because wrong VGA. Trying to install the VGA from the CD didn't work as existing VBox version detected (I think installed via `apt-get`). This is fixed when doing
```
 vagrant plugin install vagrant-vbguest
```
first, as this will then fix the VGA version on the VM. (Note: after fixing the VGA, it first warns about inconsistency of VGA versions reported by the host and guest, but this warning disappears when starting the VM again, and everything seems to works fine).

Note tha tthis xenial box has an Ubuntu user, which should be hidden/removed.

File: vagrant/Vagrantfile (hunk 2 above)
[the remark this answers is not in the document]
Comment (paskino) [reply vagrant/Vagrantfile]: yes, I've checked my tree and indeed has the `clipboard` command on a different `vb.customize` statement.

File: vagrant/Vagrantfile (hunk 3 above)
Comment (paskino) [reply vagrant/Vagrantfile]: yes

File: vagrant/Vagrantfile (hunk 1 above)
Comment (paskino) [reply vagrant/Vagrantfile]: I don't know. I guess I feel safer.

**Comment by KrisThielemans** [comment]
seems that the VM created with this box still has US keyboard (no others installed)

**Pull request closed without merging**